src/discussions/discussions-home/DiscussionsHome.test.jsx

```diff
@@ -105,4 +128,123 @@ describe('DiscussionsHome', () => {
+    axiosMock.onGet(getDiscussionsConfigUrl(courseId)).reply(200, {
+      enableInContext: true, provider: 'openedx', hasModerationPrivileges: true,
+    });
+    await executeThunk(fetchCourseConfig(courseId), store.dispatch, store.getState);
+    axiosMock.onGet(getThreadsApiUrl())
+      .reply(() => {
+        const threadAttrs = { previewBody: 'thread preview body' };
+        return [200, Factory.build('threadsResult', {}, {
+          topicId: 'noncourseware-topic-1',
+          threadAttrs,
+          count: 3,
+        })];
+      });
+    await executeThunk(fetchThreads(courseId), store.dispatch, store.getState);
+    await renderComponent(`/${courseId}/${searchByEndPoint}`);
+
+    expect(screen.queryByText(result)).toBeInTheDocument();
+  });
+
+  it.each([
+    { searchByEndPoint: 'category/section-topic-1' },
+    { searchByEndPoint: 'topics' },
+  ])('should display No Topic selected message on inContext topic pages when user has yet to select a topic %s',
+    async ({ searchByEndPoint }) => {
+      axiosMock.onGet(getDiscussionsConfigUrl(courseId)).reply(200, {
+        enableInContext: true, provider: 'openedx', hasModerationPrivileges: true,
+      });
+      await executeThunk(fetchCourseConfig(courseId), store.dispatch, store.getState);
+      axiosMock.onGet(`${getCourseTopicsApiUrl()}${courseId}`)
+        .reply(200, (Factory.buildList('topic', 1, null, {
+          topicPrefix: 'noncourseware-topic',
+          enabledInContext: true,
+          topicNamePrefix: 'general-topic',
+          usageKey: '',
+          courseware: false,
+          discussionCount: 1,
+          questionCount: 1,
+        }).concat(Factory.buildList('section', 2, null, { topicPrefix: 'courseware' })))
+          .concat(Factory.buildList('archived-topics', 2, null)));
+      await executeThunk(fetchCourseTopicsV3(courseId), store.dispatch, store.getState);
+      await renderComponent(`/${courseId}/${searchByEndPoint}`);
+
+      expect(screen.queryByText('No topic selected')).toBeInTheDocument();
+    });
+
+  it('should display empty page message for empty learners list', async () => {
+    axiosMock.onGet(getDiscussionsConfigUrl(courseId)).reply(200, {
+      learners_tab_enabled: true,
+    });
+    await executeThunk(fetchCourseConfig(courseId), store.dispatch, store.getState);
+    await renderComponent(`/${courseId}/learners`);
+
+    expect(screen.queryByText('Nothing here yet')).toBeInTheDocument();
+  });
+
+  it('should display post editor form when click on add a post button for posts', async () => {
+    await executeThunk(fetchCourseConfig(courseId), store.dispatch, store.getState);
+    await renderComponent(`/${courseId}/my-posts`);
+    await act(async () => {
+      fireEvent.click(screen.queryByText('Add a post'));
+    });
+
+    await waitFor(() => expect(container.querySelector('.post-form')).toBeInTheDocument());
+  });
+
+  it('should display post editor to add post for legacy topic view', async () => {
```

Comment:
`should display post editor to add post for legacy topic view` => `should display post editor form when click on add a post button in legacy topics view`